lib/src/components/NxList/NxListButton.tsx

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2019-present Sonatype, Inc.
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at https://www.eclipse.org/legal/epl-2.0/.
+ */
+import { faAngleRight } from '@fortawesome/free-solid-svg-icons';
+import classnames from 'classnames';
+import React, { forwardRef } from 'react';
+import { NxFontAwesomeIcon } from '../..';
+import {NxListButtonProps, nxListButtonPropTypes} from './types';
+import { includesDisabledClass } from '../../util/classUtil';
+
+const NxListButton = forwardRef<HTMLButtonElement, NxListButtonProps>(
+    function nxListButton({ children, className, disabled, selected, ...attrs}, ref) {
+      const buttonClassNames = classnames('nx-list__btn', className, {'selected': selected}, {'disabled': disabled});
```

Comment:
It should be documented that the className for this component ends up on the button, not the `li`.  I'm also wondering if we need to provide the capability to add extra classes to the `li` as well.  And I'm wondering the same thing about arbitrary extra attrs, which you currently are assigning to the button.

Reply:
> It should be documented that the className for this component ends up on the button, not the `li`. I'm also wondering if we need to provide the capability to add extra classes to the `li` as well. And I'm wondering the same thing about arbitrary extra attrs, which you currently are assigning to the button.

I'm a bit confused on how to properly accomplish this and need some guidance.

For capability to add independent classes to the `li` and `button`, do we use, for example, an `liClassName` prop that maps to the `li`'s `className` and a `buttonClassName` that maps to the `button`'s `className`?

And for extra attrs, I'm not sure how adding them to both the li and button would work. Would it make more sense to add the attrs only to the li and have the ref map to the `li` instead of the `button`?

Reply:
> I'm a bit confused on how to properly accomplish this and need some guidance.
> 
> For capability to add independent classes to the li and button, do we use, for example, an liClassName prop that maps to the li's className and a buttonClassName that maps to the button's className?
> 
> And for extra attrs, I'm not sure how adding them to both the li and button would work. Would it make more sense to add the attrs only to the li and have the ref map to the li instead of the button?

IMO you want `className` and any extra attrs to apply to one of the two underlying DOM elements, and then make custom props to enable passing of classes and attrs to the other underlying DOM element.  So for instance `className` could set extra classes on the `<li>`, and extra `NxListButtonProps` could be a subtype of `LiHTMLAttributes<HtmlLiElement>` which get passed to the `<li>`.  And then you could also have `buttonClassName` which gets included as classes on the `<button>` and `buttonAttributes` which gets included in the props for the `<button>`

Reply:
> > I'm a bit confused on how to properly accomplish this and need some guidance.
> > For capability to add independent classes to the li and button, do we use, for example, an liClassName prop that maps to the li's className and a buttonClassName that maps to the button's className?
> > And for extra attrs, I'm not sure how adding them to both the li and button would work. Would it make more sense to add the attrs only to the li and have the ref map to the li instead of the button?
> 
> IMO you want `className` and any extra attrs to apply to one of the two underlying DOM elements, and then make custom props to enable passing of classes and attrs to the other underlying DOM element. So for instance `className` could set extra classes on the `<li>`, and extra `NxListButtonProps` could be a subtype of `LiHTMLAttributes<HtmlLiElement>` which get passed to the `<li>`. And then you could also have `buttonClassName` which gets included as classes on the `<button>` and `buttonAttributes` which gets included in the props for the `<button>`

This has been implemented as suggested. Gallery documentation has been updated as well.
1248b66c8c8a03b10c3f514f9af12230277cedb5